tests/integration/utils/flow_tools.py

```diff
@@ -102,31 +106,26 @@ async def _run_flow(
     schd: Scheduler,
     level: int = logging.INFO
 ):
-    """Start a scheduler and set it running."""
+    """Start a scheduler and set the main loop running."""
     if caplog:
         caplog.set_level(level, CYLC_LOG)
 
-    # install
     await schd.install()
 
-    # start
+    task: Optional[asyncio.Task] = None
     try:
         await schd.start()
-    except Exception as exc:
-        async with timeout(5):
-            await schd.shutdown(exc)
-    # run
-    try:
+        # Do not await as we need to yield control to the main loop:
         task = asyncio.create_task(schd.run_scheduler())
-        yield caplog
 
-    # stop
     finally:
+        yield caplog
+        # Cleanup: need to shut down, but time out in case something goes wrong
         async with timeout(5):
-            # ask the scheduler to shut down nicely
+            # ask the scheduler to shut down nicely, let main loop handle it:
             schd._set_stop(StopMode.REQUEST_NOW_NOW)
-            await task
-
+            if task:
+                await task
```

Comment:
If the workflow failed to start then the main loop will not be running. This is why it separates the errors and calls `schd.shutdown`.

Reply:
The trouble is that
```python
try:
    schd.start()
except Exception as exc:
```
was making it impossible to test the Scheduler's error handling